fhirpath/src/commonMain/kotlin/com/google/fhir/fhirpath/operators/Equality.kt

```diff
@@ -25,7 +25,62 @@ import com.google.fhir.model.r4.FhirDate
 import com.google.fhir.model.r4.Quantity
 
 /** See [specification](https://hl7.org/fhirpath/N1/#equals). */
-internal fun equals(left: Any, right: Any): Boolean? {
+internal fun equal(
+  left: Collection<Any>,
+  right: Collection<Any>,
+): Boolean? {
+  if (left.isEmpty() || right.isEmpty()) {
+    return null
+  }
+  if (left.size != right.size) {
+    return false
+  }
+
+  // The FHIRPath specification states every pair of items must be equal for two collections to be
+  // equal. However, it does not explicitly states how to account for item comparison results that
+  // are empty sets. This can be interpreted as an inconsistency since this definition does not
+  // reduce nicely to the special case of collections with a single item. Consider two collections
+  // with a single item in each that are incomparable (returns empty set if compared), if we use
+  // the definition of equality for collections, we would conclude the two collections are not
+  // equal. But the special case states that an empty set should be returned.
+  // This nuance results in inconsistent behaviors in different implementations. See
+  // [discussion](https://chat.fhir.org/#narrow/channel/179266-fhirpath/topic/Collection.20equality/with/540473873).
+  // In our implementation we choose to treat empty sets as inconclusive, and return an empty set
+  // if the equality of two collections cannot be decided.
+  // Also see: https://jira.hl7.org/browse/FHIR-53076
```

Comment:
I believe that part of these statements should be included in the function kdoc, or we should move it entirely.